modvendor/modvendor_manifest_test.go

```diff
@@ -0,0 +1,386 @@
+			manifests: []manifestConfig{
+				{
+					path: "/manifest.tm",
+					patterns: []string{
+						"/file",
+					},
+				},
+			},
+			wantFiles: []string{
+				"/file",
+			},
+		},
+		{
+			name: "double-star match against directory",
+			files: []string{
+				"/b/test2/main.tf",
+				"/a/test2/main.tf",
+				"/a/b/c/test2/main.tf",
+			},
+			manifests: []manifestConfig{
+				{
+					path: "/manifest.tm",
+					patterns: []string{
+						"**/test2/main.tf",
+					},
+				},
+			},
+			wantFiles: []string{
+				"/a/b/c/test2/main.tf",
+				"/a/test2/main.tf",
+				"/b/test2/main.tf",
+			},
+		},
+		{
+			name: "double-star match against multiple directory",
+			files: []string{
+				"/b/test/test2/main.tf",
+				"/a/test/test3/main.tf",
+				"/a/b/test/c/main.tf",
+				"excluded.tf",
+			},
+			manifests: []manifestConfig{
+				{
+					path: "/manifest.tm",
+					patterns: []string{
+						"**/test/**/main.tf",
+					},
+				},
+			},
+			wantFiles: []string{
+				"/a/b/test/c/main.tf",
+				"/a/test/test3/main.tf",
+				"/b/test/test2/main.tf",
+			},
+		},
+		{
+			name: "filter patterns and dirs and files",
+			files: []string{
+				"/main.tf",
+				"/vars.tf",
+				"/fun.tf",
+				"/README.md",
+				"/LICENSE",
+				"/examples/1/main.tf",
+				"/examples/2/main.tf",
+				"/test/1/main.tf",
+				"/test/2/main.tf",
+				"/other/ohno.txt",
+			},
+			manifests: []manifestConfig{
+				{
+					path: "/manifest.tm",
+					patterns: []string{
+						"/*.tf",
+						"/README.*",
+						"/LICENSE",
+						"examples",
+					},
+				},
+			},
+			wantFiles: []string{
+				"/LICENSE",
+				"/README.md",
+				"/examples/1/main.tf",
+				"/examples/2/main.tf",
+				"/fun.tf",
+				"/main.tf",
+				"/vars.tf",
+			},
+		},
+		{
+			name: "filter config on .terramate",
+			files: []string{
+				"/main.tf",
+				"/vars.tf",
+				"/fun.tf",
+				"/README.md",
+				"/LICENSE",
+				"/examples/1/main.tf",
+				"/examples/2/main.tf",
+				"/test/1/main.tf",
+				"/test/2/main.tf",
+				"/other/ohno.txt",
+			},
+			manifests: []manifestConfig{
+				{
+					path: ".terramate/manifest.tm",
+					patterns: []string{
+						"/*.tf",
+						"/README.*",
+						"/LICENSE",
+						"examples",
+					},
+				},
+			},
+			wantFiles: []string{
+				"/LICENSE",
+				"/README.md",
+				"/examples/1/main.tf",
+				"/examples/2/main.tf",
+				"/fun.tf",
+				"/main.tf",
+				"/vars.tf",
+			},
+		},
+		{
+			name: ".terramate has precedence over root for manifest config",
+			files: []string{
+				"/file1",
+				"/file2",
+			},
+			manifests: []manifestConfig{
+				{
+					path:     "manifest.tm",
+					patterns: []string{"/file1"},
+				},
+				{
+					path:     ".terramate/manifest.tm",
+					patterns: []string{"/file2"},
+				},
+			},
+			wantFiles: []string{
+				"/file2",
+			},
+		},
+	}
+
+	for _, tcase := range testcases {
+		t.Run(tcase.name, func(t *testing.T) {
+			repoSandbox := sandbox.New(t)
+
+			// Remove the default README.md created by the sandbox
```

Comment:
I think we should get rid of this, right? I used this in the past just to create the first commit and then create the HEAD and main ref.. but we can just add another commit removing it and then all new repos are created clean.
Dont need to change now. Just documenting so we can remember of doing this next time we touch the sandbox.

Reply:
Agreed, would be cool to remove it, I was quite surprised/puzzled and it took me some time to remember this :laughing: